src/functional/merge.js

```diff
@@ -68,11 +71,22 @@ export function mergeWithSources(collection, sources, merger) {
 
 function deepMergerWith(merger) {
   function deepMerger(oldValue, newValue, key) {
-    return isDataStructure(oldValue) && isDataStructure(newValue)
+    return isDataStructure(oldValue) &&
+      isDataStructure(newValue) &&
+      areMergeable(oldValue, newValue)
       ? mergeWithSources(oldValue, [newValue], deepMerger)
       : merger
       ? merger(oldValue, newValue, key)
       : newValue;
   }
   return deepMerger;
 }
+
+function areMergeable(oldDataStructure, newDataStructure) {
```

Comment:
Nit: it would still be nice to include a comment block above this function explaining why this implementation determines things are considered mergeable

Reply:
Done. Let me know if that's what you were thinking or if you were asking for a different explanation.

Reply:
Yeah this looks good to me. It's the connection between the collection categorization and this somewhat nuanced boolean logic that will be helpful when we look at this function again after our collective memory is lost